slack_global_report.py

```diff
@@ -0,0 +1,241 @@
+        return blocks
+
+    def days(self, ur, pur):
+        blocks = []
+        header = "*Activity Per Day*"
+        blocks.append(self.sf.text_block(header))
+        uwd = ur['user_weekday']
+        days = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday']
+        text = ""
+        for idx, day in enumerate(days):
+            match = uwd.get(str(idx), [0,0])
+            m = match[0]
+            w = match[1]
+            it = "*{}* *{:,}* words in *{:,}* messages\n".format(day, w, m)
+            text += it
+        blocks.append(self.sf.text_block(text))
+        blocks.append(self.sf.divider())
+        return blocks
+
+    def hours(self, ur, pur):
+        blocks = []
+        header = "*Activity Per Hour (on Weeekdays)*"
+        blocks.append(self.sf.text_block(header))
+        hours = ur['weekday_activity_percentage']
+        text = ""
+        for hour in hours:
+            it = "*{}* *{:.2f}%*\n".format(hour, hours[hour])
+            text += it
+        blocks.append(self.sf.text_block(text))
+        blocks.append(self.sf.divider())
+        return blocks
+
+    def reacji(self, ur, pur):
+        blocks = []
+        header = "*Top Ten Reacji*"
+        blocks.append(self.sf.text_block(header))
+        reacjis = ur['top_ten_reactions']
+        text = ""
+        for reacji in reacjis:
+            it = ":{}: *{:,}*\n".format(reacji, ur['reaction'][reacji])
+            text += it
+        blocks.append(self.sf.text_block(text))
+        blocks.append(self.sf.divider())
+        return blocks
+
+    def make_report(self, ur, pur):
+        blocks = []
+        blocks += self.make_header(ur, pur)
+        blocks += self.top_channels(ur, pur)
+        blocks += self.top_users(ur, pur)
+        blocks += self.timezones(ur, pur)
+        blocks += self.days(ur, pur)
+        blocks += self.hours(ur, pur)
+        blocks += self.reacji(ur, pur)
+        blocks += self.reacted_messages(ur)
+        blocks += self.replied_messages(ur)
+        return blocks
+
+    def reacted_messages(self, ur):
+        return self.sf.messager(
+            ur['reacted_messages'],
+            'reactions',
+            show_user=True,
+            show_channel=True)
+
+    def replied_messages(self, ur):
+        return self.sf.messager(
+            ur['replied_messages'],
+            'replies',
+            show_user=True,
+            show_channel=False)
+
+    def send_report(self, ur, previous, send=True, destination=None, summary=False):
+        enricher.Enricher(fake=self.fake).enrich(ur)
+        enricher.Enricher(fake=self.fake).enrich(previous)
+        blocks = self.make_report(ur, previous)
+        if not send:
+            print("Saving report to slack.json")
+            f = open("slack.json", "w")
+            f.write(json.dumps(blocks, indent=4))
+            f.close()
+        # If set to true, this message will be sent as the user who owns the token we use
+        as_user = False
```

Comment:
Is it possible to send this as anyone besides Slack API Tester and you?

Reply:
Yeah, it's whoever has the Slack API key.  Nothing for us to do here.